packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1329,7 +1329,7 @@ function commitSuspenseComponent(finishedWork: Fiber) {
       if (thenables !== null) {
         suspenseCallback(new Set(thenables));
       }
-    } else if (__DEV__) {
+    } else if (__DEV__ && suspenseCallback != null) {
```

Comment:
We usually check undefined and null separately (partly to avoid document.all being considered).